php-classes/Slate/CBL/Tasks/StudentTasksRequestHandler.php

```diff
@@ -50,13 +54,113 @@ public static function handleAssignedRequest($options = [], $conditions = [])
 
     }
 
+    public static function handleStudentTasksListRequest()
+    {
+        $student = static::_getRequestedStudent();
+        $courseSection = static::_getRequestedCourseSection();
+        $courseSectionIds = [];
+
+        if (!$courseSection) {
+            $currentTerm = Term::getCurrent();
+            $sections = [];
+
+            if ($currentTerm) {
+                $sections = Section::getAllByField('TermID', $currentTerm->ID);
+            }
+
+            if ($currentTerm && count($sections) > 0) {
+                $courseSectionIds = array_map(function($s) {
+                    return $s->ID;
+                }, $sections);
+            } else {
+                // If there was no section sent, and there are no sections in the current term, there is no reason to run a query
+                return static::respond('tasks',[
+                    'success' => true
+                    ,'data' => []
+                    ,'total' => 0
+                ]);
+            }
+        }
+
+        $query = '
+            SELECT Tasks.ID as TaskID,
+                   Tasks.ParentTaskID as ParentID,
+                   Tasks.Title as TaskTitle,
+                   StudentTasks.StudentId as StudentID,
+                   StudentTasks.ID as ID,
+                   StudentTasks.TaskStatus as TaskStatus,
+                   UNIX_TIMESTAMP(StudentTasks.DueDate) as DueDate,
+                   CourseSections.Title as CourseSectionTitle,
+                   CourseSections.Code as CourseSectionCode,
+                   UNIX_TIMESTAMP(Submissions.UltimateSubmission) as SubmittedDate
+            FROM %1$s StudentTasks
+            JOIN %2$s Tasks on Tasks.ID = StudentTasks.TaskID
+            JOIN %3$s CourseSections on CourseSections.ID = StudentTasks.CourseSectionID
+            LEFT JOIN (
+                SELECT StudentTaskID as SubmissionStudentTaskID,
+                       MAX(Created) as UltimateSubmission
+                FROM %4$s
+                GROUP BY SubmissionStudentTaskID
+            ) as Submissions ON (SubmissionStudentTaskID = StudentTasks.ID )
+            WHERE StudentTasks.StudentID = %5$d
+        ';
+
+        if ($courseSection) {
+            $query .= ' AND StudentTasks.CourseSectionID = '.$courseSection->ID;
+        } else {
+            $query .= ' AND StudentTasks.CourseSectionID IN ('.implode(',',$courseSectionIds).')';
+        }
+
+        /* In the student task dashboard, tasks should be organized by status and then sorted by due date.

```

Comment:
@BillClinton this is a clue that this code should be in the student task dashboard request handler. One of the goals for the optimization is to eliminate any specialty routes that have been added to the CBL `RecordsRequestHandler` endpoints so that all that is really added to them is custom query string filters for `handleBrowseRequest`, and they always return the model their built for.

Any endpoints you need that don't return a model are probably returning something custom that's designed for a particular UI, and should be contained in the UI-oriented *DashboardRequestHandler endpoints